src/asm.rs

```diff
@@ -611,114 +611,126 @@ fn estimate_template_length(
 }
 
 /// Converts a register class to a GCC constraint code.
-fn reg_to_gcc(reg: InlineAsmRegOrRegClass) -> ConstraintOrRegister {
-    let constraint = match reg {
-        // For vector registers LLVM wants the register name to match the type size.
```

Comment:
this comment is lost right now, I'm not sure what it was supposed to say, and why what llvm does is relevant?

Reply:
Oh, that was copy-pasted from the LLVM codegen, but I forgot to change that to GCC.

Reply:
ok, but what does this comment mean? What type is talked about here? it just doesn't seem like the right place for this comment?

Reply:
It comes from [here](https://github.com/rust-lang/rust/blob/6ac88786155774da429213a1362d4cb085047af3/compiler/rustc_codegen_llvm/src/asm.rs#L587) originally.
Maybe the type is the `layout` parameter that we don't have in cg_gcc.

Reply:
yeah I think that's right, e.g. 

```rust
  match layout.size.bytes() {
                        64 => 'z',
                        32 => 'y',
                        _ => 'x',
```

this really does pick the register width based on a type. Do you think GCC will have to do the same? So far that does not seem necessary.

Reply:
I do not know. I would need to check if needed.